### DISCUSSION TITLE
How to use reducer combined with storage

### DISCUSSION THREAD

**Discussion by tiamed** [pmndrs/jotai#1306] [Q&A] [answered]
Is it possible to use reducer with storage?
Will atomWithReducer support derived atom as its first parameter?

**Comment by tiamed** [comment]
> Is it possible to use reducer with storage? Will atomWithReducer support derived atom as its first parameter?

Found useReducerAtom which solved this question.

**Comment by dai-shi** [comment, marked as answer]
`atomWithFoo` can't derive from different atoms.
FYI: https://twitter.com/dai_shi/status/1550803662292779008

Related discussions: #1304, #933 (and some more in Discord servers)

**Comment by vishal-vp** [comment]
From the official docs: https://jotai.org/docs/guides/composing-atoms#extending-atoms-with-atomwith-utils
